src/timer.rs

```diff
@@ -4,18 +4,107 @@
 // TIM1 is 16 bit.
 
 use crate::hal::timer::{CountDown, Periodic};
-use crate::stm32::{TIM1, TIM2};
+use crate::stm32::{LPTIM1, LPTIM2, LPTIM3, LPTIM4, LPTIM5};
+use crate::stm32::{
+    TIM1, TIM12, TIM13, TIM14, TIM15, TIM16, TIM17, TIM2, TIM3, TIM4, TIM5,
+    TIM6, TIM7, TIM8,
+};
 
 use cast::{u16, u32};
 use nb;
 use void::Void;
 
-use crate::rcc::{CoreClocks, APB1, APB2};
+use crate::rcc::Ccdr;
+use crate::rcc::{APB1, APB2};
+use crate::stm32::rcc::{d2ccip2r, d3ccipr};
 use crate::time::Hertz;
 
+/// Associate clocks with timers
+pub trait GetClk {
```

Comment:
This sounds like it would benefit from being and is generic enough to be moved into `embedded-hal`. Together with the `Ccdr` construction.